SonarQube.Bootstrapper/BootstrapperSettings.cs

```diff
@@ -44,28 +58,34 @@ public class BootstrapperSettings : IBootstrapperSettings
         public const string RelativePathToTempDir = @".sonarqube";
         public const string RelativePathToDownloadDir = @"bin";
 
-        private readonly ILogger logger;
 
+        #endregion
+
+        #endregion
+
+        private readonly ILogger logger;
         private readonly string sonarQubeUrl;
+        private readonly AnalysisPhase analysisPhase;
+
         private string tempDir;
         private string preProcFilePath;
         private string postProcFilePath;
-
+        
         #region Constructor(s)
         
-        public BootstrapperSettings(string sonarQubeUrl, ILogger logger)
+        public BootstrapperSettings(string sonarQubeUrl, AnalysisPhase phase, ILogger logger)
         {
-            if (string.IsNullOrWhiteSpace(sonarQubeUrl))
+            if (sonarQubeUrl == null)
```

Comment:
In the v0.9 post-processing case, the bootstrapper won't be passed any dynamic properties or the location of a settings file so it won't directly be able to work out what the URL is, which is why this argument check now accepts an empty string.
I'll have another look at how the settings are constructed as part of SONARMSBRU-81: "Do not forward all properties to the generated sonar-runner.properties file"